### PR TITLE
feat: onboarding router batching for braze

### PR DESCRIPTION
**Pull request by utsabc**
## Onboarding router batching for braze

> Added support for batching in router transform for braze

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -299,9 +306,77 @@ const processDeduplication = (userStore, payload) => {
...
+    eventsArrayChunks.length,
+    eventsArrayChunks.length,
+  );
+  const responseArray = [];
+  for (let i = 0; i < maxNumberOfRequest; i += 1) {
+    const attributes = attributeArrayChunks[i];
+    const events = eventsArrayChunks[i];
+    const purchases = purchaseArrayChunks[i];
+    const response = defaultRequestConfig();
+    response.endpoint = getTrackEndPoint(getEndpointFromConfig(destination));
```
Comment (sanpj2292) [inline src/v0/destinations/braze/util.js:348]: [Q] Will `getEndpointFromConfig` change wrt the event ? I mean can we call this method outside the loop and make use of the value from there ?!

File: src/v0/destinations/braze/util.js [hunk 2]
```diff
@@ -299,9 +306,77 @@ const processDeduplication = (userStore, payload) => {
...
+    }
+  }
+  const attributeArrayChunks = _.chunk(attributesArray, 75);
+  const eventsArrayChunks = _.chunk(eventsArray, 75);
+  const purchaseArrayChunks = _.chunk(purchaseArray, 75);
+
+  const maxNumberOfRequest = Math.max(
+    attributeArrayChunks.length,
+    eventsArrayChunks.length,
+    eventsArrayChunks.length,
```
Comment (sanpj2292) [inline src/v0/destinations/braze/util.js:340]: This should be `purchaseArrayChunks` right ?

File: src/v0/destinations/braze/util.js [hunk 3]
```diff
@@ -299,9 +306,77 @@ const processDeduplication = (userStore, payload) => {
...
+  );
+  const responseArray = [];
+  for (let i = 0; i < maxNumberOfRequest; i += 1) {
+    const attributes = attributeArrayChunks[i];
+    const events = eventsArrayChunks[i];
+    const purchases = purchaseArrayChunks[i];
```
Comment (sanpj2292) [inline src/v0/destinations/braze/util.js:346]: When maxNumberOfRequest for attributes and events is greater than purchases, `purchaseArrayChunks[i]` would be `undefined` at some point, are we ok with sending `undefined` in `purchases` object to braze ? 🤔

Comment (sanpj2292) [reply src/v0/destinations/braze/util.js:346]: May be we should send `[]` to braze, but if we need to check if sending `undefined` for purchases, attributes or events is ok

Comment (utsabc) [reply src/v0/destinations/braze/util.js:351]: It should get handled by `removeUndefinedAndNullValues` in line 355. I will write test cases for this though

File: src/v0/destinations/braze/util.js (hunk 2 above)
Comment (utsabc) [reply src/v0/destinations/braze/util.js]: Addressed

File: src/v0/destinations/braze/util.js (hunk 1 above)
Comment (utsabc) [reply src/v0/destinations/braze/util.js]: Yep.handled

File: src/v0/destinations/braze/util.js (hunk 3 above)
Comment (sanpj2292) [reply src/v0/destinations/braze/util.js:352]: Perfect. Please write test-cases for this and feel free to resolve the comments

**Review by sanpj2292 — APPROVED** [review]
LGTM

**Review by koladilip — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**